WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageListDialogHelper.kt

```diff
@@ -0,0 +1,89 @@
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.UNPUBLISHED_REVISION_DIALOG_LOAD_UNPUBLISHED_VERSION_CLICKED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.UNPUBLISHED_REVISION_DIALOG_SHOWN
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.RemoteId
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.ui.posts.PostUtils
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.viewmodel.helpers.DialogHolder
+
+private const val CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG = "CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG"
+private const val CONFIRM_DELETE_PAGE_DIALOG_TAG = "CONFIRM_DELETE_PAGE_DIALOG_TAG"
+
+class PageListDialogHelper(
+    private val showDialog: (DialogHolder) -> Unit,
+    private val analyticsTracker: AnalyticsTrackerWrapper
+) {
+    private var pageIdForAutosaveRevisionResolutionDialog: RemoteId? = null
+    private var pageIdForDeleteDialog: RemoteId? = null
+
+    fun showAutoSaveRevisionDialog(page: PostModel) {
+        analyticsTracker.track(UNPUBLISHED_REVISION_DIALOG_SHOWN)
+        val dialogHolder = DialogHolder(
+                tag = CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG,
+                title = UiStringRes(R.string.dialog_confirm_autosave_title),
+                message = PostUtils.getCustomStringForAutosaveRevisionDialog(page, true),
+                positiveButton = UiStringRes(R.string.dialog_confirm_autosave_restore_button),
+                negativeButton = UiStringRes(R.string.dialog_confirm_autosave_dont_restore_button)
+        )
+        pageIdForAutosaveRevisionResolutionDialog = RemoteId(page.remotePostId)
+        showDialog.invoke(dialogHolder)
+    }
+
+    fun showDeletePageConfirmationDialog(pageId: RemoteId, pageTitle: String) {
+        val dialogMessage = {
+            WordPress.getContext().getString(R.string.page_delete_dialog_message, pageTitle)
+        }
+
+        val dialogHolder = DialogHolder(
+                tag = CONFIRM_DELETE_PAGE_DIALOG_TAG,
+                title = UiStringRes(R.string.delete_page),
+                message = UiStringText(dialogMessage()),
+                positiveButton = UiStringRes(R.string.delete),
+                negativeButton = UiStringRes(R.string.cancel)
+        )
+        pageIdForDeleteDialog = pageId
+        showDialog.invoke(dialogHolder)
+    }
+
+    fun onPositiveClickedForBasicDialog(
+        instanceTag: String,
+        deletePage: (RemoteId) -> Unit,
+        editPage: (RemoteId, LoadAutoSaveRevision) -> Unit
+    ) {
+        when (instanceTag) {
+            CONFIRM_DELETE_PAGE_DIALOG_TAG -> pageIdForDeleteDialog?.let {
+                pageIdForDeleteDialog = null
+                deletePage(it)
+            }
```

Comment:
Resolved in ca2ccdc

Reply:
Is there any advantage using `run` instead of simply calling ```?: throw NullPointerException("pageIdForAutosaveRevisionResolutionDialog shouldn't be null.")```?

Reply:
Nope! Thanks! I sometimes forget these syntax/language tricks 😅  Resolved in 2a6afe5